resources/faq.md

```diff
@@ -0,0 +1,162 @@
+
+## I am trying to add my documents but I keep receiving a `400 - Invalid data` response.
+
+MeiliSearch API only accepts JSON format.<br>
+In case of a [document addition](/references/documents.md#add-or-replace-documents), only arrays of objects are expected.
+
+The `400 - Invalid data` response probably means that your data is not in an expected format.
+
+Most common errors:
+- Extraneous comma at the end of a line.
+- Data is not an array of objects: for the [document addition route](/references/documents.md#add-or-replace-documents), MeiliSearch only accepts an array in the body even if there is only one document.
+
+Wrong:
+```json
+{ "id": 123,  "title": "Pride and Prejudice" }
+```
+
+Good:
+```json
+[
+    { "id": 123,  "title": "Pride and Prejudice" }
+]
+```
+
+:::tip
+The [jq](https://github.com/stedolan/jq) command line tool can greatly help you check the format of your data.
+```bash
+$ cat your_file.json | jq
+```
+:::
+
+## My document upload failed with the `document id is missing` error.
+
+::: note TLDR;
+Most common reasons:
+- A unique identifier in your document is missing.
+- The unique identifier of your document is not well-formatted.
+:::
+
+Each document is required to contain a unique identifier. This identifier attribute is the `primary key`.
+
+How do I know the primary key of my index? [Check this route](/references/indexes.md#get-one-index). The `null` value means it has not been defined yet.
+
+By default, the primary key will be inferred in the first document sent. MeiliSearch will search for an attribute that contains the string `id` in a case-insensitive manner (e.g., `uid`, `MovieId`, `ID`, `123id123`). If none has been found, no documents will be added.
+
+If you get a `document id is missing` error, the primary key was not recognized. This means your primary key is wrongly formatted. Sending primary key's name as a query parameter [when adding documents](/references/documents.md#add-or-replace-documents) should solve this issue.
+
+Note that the primary key value must contain only `A-Z a-z 0-9` and `-_` characters.
+
+Wrong:
+```json
+"id": "@BI+* ^5h2%"
+```
+
+Good:
+```json
+"id": "_Aabc012_"
+```
+
+See more elaborate [information about the primary key](/guides/main_concepts/documents.md#primary-key).
+
+## I have uploaded my documents, but I get no result when I search in my index.
+
+Your documents upload probably failed.<br>
+To understand what happened, please check this [answer](/resources/faq.md#i-did-a-call-to-an-api-route-and-i-only-got-an-updateid-as-a-response-what-does-it-mean).
+
+## Does MeiliSearch deliver an interface to search in my documents?
+
+Yes, a web interface is available on the default address and port of your MeiliSearch instance.
+
+All you need to do is open your web browser and enter MeiliSearch’s address to visit it locally. This will lead you to a web page with a search bar that will allow you to search in a selected index.
```

Comment:
I just copy/past your article 😂

```
all you need to do is open your web browser and enter MeiliSearch’s address to visit it locally. This will lead you to a web page with a search bar that will allow you to search in a selected index.
```